apis/v1beta1/shared_types.go

```diff
@@ -130,15 +148,19 @@ type CommonRouteSpec struct {
 	// to be attached to. Note that the referenced parent resource needs to
 	// allow this for the attachment to be complete. For Gateways, that means
 	// the Gateway needs to allow attachment from Routes of this kind and
-	// namespace.
+	// namespace. For Services, that means the Service must either be in the same
+	// namespace for a "producer" route, or the mesh implementation must support
+	// and allow "consumer" routes for the referenced Service.
```

Comment:
I may have missed this, could someone use a ReferenceGrant to turn a Producer route into a Consumer route? Are Is a ReferenceGrant that grants access from `HTTPRoute` to `Service` _only_ for `BackendRefs` and not `ParentRefs`? Assuming that's the case, we probably need to very clearly state that somewhere.

Reply:
I think what we need to say here is that ReferenceGrant is not applicable for ParentRefs to Services. Then in ReferenceGrant API Spec we should probably clarify that when a ReferenceGrant is used to grant access from a Route to a Service, that is _only_ for BackendRefs, not for ParentRefs.